workflow/controller/operator.go

```diff
@@ -554,6 +554,13 @@ func (woc *wfOperationCtx) persistUpdates(ctx context.Context) {
 		woc.controller.hydrator.HydrateWithNodes(woc.wf, nodes)
 	}
 
+	// The workflow returned from wfClient.Update doesn't have a TypeMeta associated
+	// with it, so copy from the original workflow.
+	woc.wf.TypeMeta = woc.orig.TypeMeta
+
+	// Create WorkflowNode* events for nodes that have changed phase
+	woc.recordNodePhaseChangeEvents(&woc.orig.Status.Nodes, &woc.wf.Status.Nodes)
```

Comment:
Wfv1.Nodes is a map, which can be nil. You don’t need to use pointer references here. That will simplify the the func too.

Reply:
Ah, I didn't realize you were suggesting changing the method signature before; that makes sense. Updated in https://github.com/argoproj/argo-workflows/pull/5531/commits/2b302e8fe7f3e4cbbfbb1219f76cca7af3f9c995.